src/triage/component/architect/feature_group_mixer.py

```diff
@@ -31,6 +32,23 @@ def leave_one_out(feature_groups):
         results.append(feature_dict)
     return results
 
+def all_combinations(feature_groups):
+    """Return all combinations of groups, excluding repeated groups
+
+    Args:
+        feature_groups (list) The feature groups to apply the strategy to
+
+    Returns: A list of feature dicts
+    """
+    results = []
+    for number_feature_groups in range(0, len(feature_groups) + 1):
+        for combo in itertools.combinations(feature_groups,\
+                                            number_feature_groups):
+            feature_dict = FeatureGroup()
+            for group_element in combo:
+                feature_dict.update(group_element)
+            results.append(feature_dict)
```

Comment:
I guess the `filter` is necessary because you might not have added any elements to the `FeatureGroup`? Perhaps better to "filter" these out eagerly?

```suggestion
            if feature_dict:
                results.append(feature_dict)
```

Reply:
@jesteria When creating the list of `feature_groups` using the new function (`all_combinations`), the function will return a list with always a empty `FeatureGroup` object, that is because the `itertooll.combination` is also taking the empty set as a possible combination (you can run the test, it made things clearer to me at least). 

Probably there's a better way of doing this and we don't need the filter, what do you think?

Reply:
When I run the test with @jesteria 's suggested code changes it still passes